Tests/LibC/CMakeLists.txt

```diff
@@ -40,6 +41,8 @@ set(TEST_SOURCES
 
 set_source_files_properties(TestMath.cpp PROPERTIES COMPILE_FLAGS "-fno-builtin")
 set_source_files_properties(TestStrtodAccuracy.cpp PROPERTIES COMPILE_FLAGS "-fno-builtin-strtod")
+# Prevent floating-point calculations from being constant folded.
+set_source_files_properties(TestFenv.cpp PROPERTIES COMPILE_FLAGS "-O0")
```

Comment:
Is there a more fine-grained optimization disable flag we can put here than O0? like -fno-constant-fold or something?

Reply:
I wonder if `-frounding-math` would solve all of our problems (including the necessity of `black_box_add`): https://godbolt.org/z/TxMEbnEGb

Reply:
I'm not sure there are performance concerns, and with half an hour of reading GCC documentation I couldn't find a flag specific to constant folding, just one for constant merging (which I think is more about deduplicating constants than evaluating static expressions).

Reply:
@DanShaders beat me to it! (VSCode didn't show your comment when I wrote my answer) Looks like you are right here, I will try this immediately.